tests/test_freqtradebot.py

```diff
@@ -4581,6 +4590,318 @@ def make_sure_its_651(*args, **kwargs):
+                 MagicMock(return_value=closed_sell_dca_order_1))
+    mocker.patch('freqtrade.exchange.Exchange.fetch_order_or_stoploss_order',
+                 MagicMock(return_value=closed_sell_dca_order_1))
+    assert freqtrade.execute_trade_exit(trade=trade, limit=ask,
+                                        sell_reason=SellCheckTuple(sell_type=SellType.STOP_LOSS),
+                                        sub_trade_amt=amount)
+    trades: List[Trade] = trade.get_open_trades_without_assigned_fees()
+    assert len(trades) == 1
+    # Assert trade is as expected (averaged dca)
+
+    trade = Trade.query.first()
+    assert trade
+    assert trade.open_order_id is None
+
+    orders = Order.query.all()
+    assert orders
+    assert len(orders) == 2
+    # Make sure the closed order is found as the second order.
+    order = trade.select_order('sell', False)
+    assert order.order_id == '601'
+
+    amount = 50
+    ask = 16
+    closed_sell_dca_order_2 = {
+        'ft_pair': pair,
+        'status': 'closed',
+        'ft_order_side': 'sell',
+        'side': 'sell',
+        'type': 'limit',
+        'price': ask,
+        'average': ask,
+        'amount': amount,
+        'filled': amount,
+        'cost': amount * ask,
+        'ft_is_open': False,
+        'id': '602',
+        'order_id': '602'
+    }
+    mocker.patch('freqtrade.exchange.Exchange.create_order',
+                 MagicMock(return_value=closed_sell_dca_order_2))
+    mocker.patch('freqtrade.exchange.Exchange.fetch_order',
+                 MagicMock(return_value=closed_sell_dca_order_2))
+    mocker.patch('freqtrade.exchange.Exchange.fetch_order_or_stoploss_order',
+                 MagicMock(return_value=closed_sell_dca_order_2))
+    assert freqtrade.execute_trade_exit(trade=trade, limit=ask,
+                                        sell_reason=SellCheckTuple(sell_type=SellType.STOP_LOSS),
+                                        sub_trade_amt=amount)
+    # Assert trade is as expected (averaged dca)
+
+    trade = Trade.query.first()
+    assert trade
+    assert trade.open_order_id is None
+    assert trade.close_profit_abs == 94.25
+
+    orders = Order.query.all()
+    assert orders
+    assert len(orders) == 3
+
+    # Make sure the closed order is found as the second order.
+    order = trade.select_order('sell', False)
+    assert order.order_id == '602'
+    assert trade.is_open is False
+
+
+def test_position_adjust3(mocker, default_conf_usdt, fee) -> None:
+
+    orders_list = [
+        (
+            # side ampunt, price
+            ('buy', 100, 10),
+            ('buy', 100, 15),
+            ('sell', 50, 12),
+            ('sell', 100, 20),
+            ('sell', 50, 5),
+        ),
+        (
+            ('buy', 100, 3),
+            ('buy', 100, 7),
+            ('sell', 100, 11),
+            ('buy', 150, 15),
+            ('sell', 100, 19),
+            ('sell', 150, 23),
+        )
+    ]
```

Comment:
I do like this concept - i'd however

* move this to a parameter (`@pytest.mark.parametrize`) so it's separate tests, which simplifies debugging should one case fail
* Add several "expected" values for each order (may also be `None` if it doesn't make sense) to check the value after each order
*  Evaluate and therefore double-check the trade's (amount, profit, realized profit, unrealized profit, ...) after each step (that's the expected values from above).

Reply:
Updated